src/averageAges.js

```diff
@@ -54,9 +69,26 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
-}
+const calculateAverageAgeDiff = (people, onlyWithSon) => {
+  const woman = people.filter(({ sex, mother }) =>
+    onlyWithSon
+      ? sex === 'm' && people.some(({ name }) => name === mother)
+      : people.some(({ name }) => name === mother)
+  );
+
+  const ageDiff = woman.map(child => {
+    const mother = people.find(({ name }) => name === child.mother);
+
+    child.motherAgeInBirth = child.born - mother.born;
+
+    return child;
+  });
+
+  const totalAges = ageDiff.reduce((sum, { motherAgeInBirth }) =>
+    sum + motherAgeInBirth, 0);
```

Comment:
```suggestion
  const totalAges = ageDiff.reduce((sum, { motherAgeInBirth }) => (
    sum + motherAgeInBirth
  ), 0);
```
It's easier to see where the callback ends, if you write it this way